packages/flutter_tools/lib/src/device.dart

```diff
@@ -93,6 +93,20 @@ class DeviceManager {
       }
     }
   }
+
+  /// Whether we're capable of listing any devices given the current environment configuration.
+  bool get canListAnything {
+    return _platformDiscoverers.every((DeviceDiscovery discoverer) => discoverer.canListAnything);
```

Comment:
Should this be `any` instead of `every`?  I guess it's a question of semantics -- if one discovery can list devices but another can't, then what should we return for `canListAnything`?

Reply:
absolutely - good catch